### PR TITLE
[16.0][ADD] helpdesk_product

### PR DESCRIPTION
**Pull request by MariaBForgeFlow**
This module adds the option to select product in the helpdesk tickets.

@ForgeFlow

### CODE REVIEW

File: helpdesk_product/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "Helpdesk Product",
+    "summary": "Add the option to select product in the tickets.",
+    "author": "ForgeFlow, " "Odoo Community Association (OCA)",
+    "version": "16.0.1.0.0",
+    "license": "AGPL-3",
+    "category": "After-Sales",
+    "website": "https://github.com/OCA/helpdesk",
+    "depends": ["helpdesk_mgmt", "sale_management"],
```
Comment (dalonsod) [inline helpdesk_product/__manifest__.py]: Is needed such dependency, shoudn't be `product` addon instead enough?

File: helpdesk_product/views/helpdesk_ticket_view.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+         <!-- SEARCH TICKET HELPDESK -->
+        <record id="helpdesk_ticket_view_search" model="ir.ui.view">
+            <field name="name">helpdesk.ticket.search</field>
+            <field name="model">helpdesk.ticket</field>
+            <field name="inherit_id" ref="helpdesk_mgmt.helpdesk_ticket_view_search" />
+            <field name="arch" type="xml">
+                <field name="user_id" position="after">
+                    <field name="product_id" />
+                </field>
+            </field>
```
Comment (dalonsod) [inline helpdesk_product/views/helpdesk_ticket_view.xml:52]: You could add some shortcuts:
* Filter for tickets with product.
* Group by product

**Review by dalonsod — CHANGES_REQUESTED** [review]
Hello @mariadforgeflow thanks for the contribution!

See comments, and also and not blocking, I'd suggest adding a product flag (in template) and/or category that indicates whether a product is available for tickets, so some product could be excluded for this functionality if needed.

**Comment by MariaBForgeFlow** [comment]
Hello @dalonsod 

I have modified the module to include the following :

- Option to make certain products unavailable for tickets
- Filter for tickets which contain associated products
- Group tickets by product

I have also changed the module's dependencies so that it depends on `"product"` instead of `"sale_management"`.

File: helpdesk_product/views/helpdesk_ticket_view.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            <field name="arch" type="xml">
+                <field name="user_id" position="after">
+                    <field name="product_id" />
+                </field>
+
+                <search>
+                    <filter
+                    string="Product Related"
+                    name="product_filter"
+                    domain="['!', ('product_id', '!=', '' )]"
```
Comment (dalonsod) [inline helpdesk_product/views/helpdesk_ticket_view.xml]: Simpler domain IMO:
```suggestion
                    domain="[('product_id', '!=', False)]"
```

File: helpdesk_product/views/helpdesk_ticket_view.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        </record>
+
+        <!-- TREE TICKET HELPDESK -->
+        <record id="ticket_view_tree" model="ir.ui.view">
+            <field name="name">helpdesk.ticket.tree</field>
+            <field name="model">helpdesk.ticket</field>
+            <field name="inherit_id" ref="helpdesk_mgmt.ticket_view_tree" />
+            <field name="arch" type="xml">
+                <field name="user_id" position="after">
+                    <field name="product_id" />
```
Comment (dalonsod) [inline helpdesk_product/views/helpdesk_ticket_view.xml]: Make it optional?

**Review by dalonsod — APPROVED** [review]
Tested on runboat :+1: 
See minor comments anyway. And I only miss row things: a "Products enabled for helpdesk" menu for Helpdesk Managers, like e.g. e-commerce enabled products, and some demo data, but they're not blocking IMO.

**Comment by MariaBForgeFlow** [comment]
> Tested on runboat 👍 See minor comments anyway. And I only miss row things: a "Products enabled for helpdesk" menu for Helpdesk Managers, like e.g. e-commerce enabled products, and some demo data, but they're not blocking IMO.

Hello David,

I don't quite understand what you mean by having a "Products enabled for helpdesk" menu. Do you mean adding a filter on the product search view to search by products that are enabled for helpdesk?

**Comment by dalonsod** [comment]
I mean something like this:

imagen

Adding a shortcut to products management within Helpdesk app like e-commerce does.

**Comment by MariaBForgeFlow** [comment]
I will look into it, thank you.

**Comment by MariaBForgeFlow** [comment]
Hello @dalonsod 

I added a menu to the products that are available for helpdesk tickets on the Helpdesk application so that they can be accessed from Helpdesk. Please let me know if there's any other functionality you find missing.

**Comment by dalonsod** [comment]
:+1:

**Review by DavidJForgeFlow — APPROVED** [review]
LGTM!

**Review by flotho — COMMENTED** [review]
Thanks for the contributoin.

Regarding the technical everything looks fine but I didn't understood the use case behing this module. 
Any chance to have an explanation of the business you're expecting with this module ?
Regards

**Comment by MariaBForgeFlow** [comment]
> Thanks for the contributoin.
> 
> Regarding the technical everything looks fine but I didn't understood the use case behing this module. Any chance to have an explanation of the business you're expecting with this module ? Regards

Hello Florent,

This module will allow helpdesk users to link a product on a helpdesk ticket to the product's record. One example of how this can be useful is for retail companies, as they might receive complaints from customers regarding specific products, and linking the product on the helpdesk ticket will provide more information to the helpdesk user.
Please let me know if there's anything else I can clarify.

**Comment by JordiBForgeFlow** [comment]
@dalonsod Can you merge this one?

**Comment by dalonsod** [comment]
I'm afraid I can't. I have not such access level in this repo 😅

**Comment by rousseldenis** [comment]
@JordiBForgeFlow @LoisRForgeFlow Could you confirm this module is to select the product concerned by the helpdesk ticket (and not a 'Helpdesk Product') ? Thanks

**Comment by LoisRForgeFlow** [comment]
@rousseldenis I was not involved in this one but looking at this comment https://github.com/OCA/helpdesk/pull/520#issuecomment-2036757249, I would say is the the product concerned by the helpdesk ticket.

**Comment by rousseldenis** [comment]
> @rousseldenis I was not involved in this one but looking at this comment [#520 (comment)](https://github.com/OCA/helpdesk/pull/520#issuecomment-2036757249), I would say is the the product concerned by the helpdesk ticket.

@LoisRForgeFlow Thanks!

**Pull request merged**